fetchmail/fetchmail.py

```diff
@@ -45,7 +44,8 @@ def run(connection, cursor):
     for line in cursor.fetchall():
         fetchmailrc = ""
         user_email, protocol, host, port, tls, username, password = line
-        options = "options ssl" if tls else ""
+        options = "options fetchall antispam 501, 504, 550, 553, 554"
```

Comment:
Agree with your concern. fetchall can be dangerous if another application is relying on the messages being there on the remote server.
Being selfish: in my setup fetchmail is the only application pulling messages, and I don't want them to remain on the remote.